shortfin/python/shortfin_apps/llm/components/messages.py

```diff
@@ -52,35 +52,22 @@ def reset(self, phase: InferencePhase):
         self.return_all_logits = False
         self.return_host_array = True
         self.result_logits = None
+        self.allocation.release_pages()
+        self.allocation = None
 
     def cache_page_indices(self, max_len: int) -> list[int]:
-        if not self.locked_pages:
+        if not self.allocation:
             return []
-        indices = [p.index for p in self.locked_pages]
-        if len(indices) > max_len:
-            return indices[0:max_len]
-        return indices
+        indices = [p.index for p in self.allocation.get_page_list()]
```

Comment:
```python
return [p.index for p in self.allocation.get_page_list()[:max_len]]
```

Saves a couple iterations to not populate entire list, then only return a subset of the list

Reply:
nice catch! implemented.